### ISSUE TITLE
can't right click map (set a from/to location) on Mac / Safari

### ISSUE THREAD

**Issue by fpurcell** [conveyal/trimet-mod-otp#75]
to do a right click on a mac, you use the (one button) mouse and the control key ... FF and Chrome on Mac work fine, but with Safari, the popup bubble appears then disappears when you let go of the mouse.  The only way to keep the bubble up is the control-mouse_click-move_the_map in one motion, then the popup stays up.

**Comment by demory** [comment]
What version of MacOS / Safari is this happening on? Seems to be working OK on my Mac.

**Comment by fpurcell** [comment]
High Sierra / Version 11.1.2 (13605.3.8) 

Have a couple of Mac's (13" Airs), and when I right click the map on Safari, the popup on the map to set a from/to location flashes on the screen, with no way to click one of the links...

**Comment by fpurcell** [comment]
fyi, about the Mac issue and right click ... just attached a 3-button mouse, and the right button works fine.  But control-click left-button (Mac's one-button mouse version of a right-click) doesn't work on the map.  I think this is a problem of medium priority; if other Mac users' brains have been  hacked to the extent of my own, control-click is something done all the time ... it's so much so that control-click _is_ right-click, and it takes conscious effort to describe it otherwise (I'm so damaged :).

**Comment by demory** [comment]
@fpurcell This one has been driving me absolutely bonkers. I've spent the day on this and do not have a resolution yet, but I will summarize what I have found.

Ctrl-left-click as a substitute for right-click on Mac is working fine on Chrome and Firefox; only Safari shows the described behavior (but I can reproduce it there). After much debugging I determined that, on Safari only, a control-click fires both a context-menu event *and* a left-click event. This is why the popup flashes on screen; the left-click event fires immediately after the menu event which has the effect of clearing the popup right away.

After some digging around, I found the following discussion, which seems to be relevant here. It turns out Webkit does fire a button=0 event (i.e. left-click) on ctrl-click and this is intentional. This was reported here as a bug, but according to the discussion is considered by some to be the proper behavior, though it is acknowledged to be different from other browsers. There seems to be some ambiguity about interpreting the HTML5 spec on this.

https://bugs.webkit.org/show_bug.cgi?id=52174

I've tried a few simple fixes, including trying to suppress the left-click event using preventDefault() when ctrl is also active, but to no avail.

I need to shift focus to some of the other priority issues, but I will come back to this -- the other options I want to explore are:
- See if Leaflet can be patched to handle this case, though that means getting into the guts of Leaflet which could be a major undertaking
- See if there is some workaround on our side that recognizes when it's on a webkit browser and adjusts map behavior accordingly

FWIW, I would be surprised if this affects a large number of users (If it did, I would expect to find at least one Leaflet ticket related to it -- there are none that I can find). I think most Mac users today use either a multi-button mouse or (increasingly) a multitouch trackpad, where right-click is performed by a two-finger tap, which does work as expected here. But I understand that supporting control-click for longtime Mac users is important.
